libc/getent/pwent.c

```diff
@@ -46,17 +46,19 @@ setpwent(void)
 void
 endpwent(void)
 {
-    if (pw_fd != -1)
+    if (pw_fd != -1) {
         close(pw_fd);
-    pw_fd = -1;
+        pw_fd = -1;
+    }
 }
 
 struct passwd *
 getpwent(void)
 {
-    if (pw_fd == -1)
+    if (pw_fd == -1) {
         setpwent();
-    if (pw_fd != -1)
-        return __getpwent(pw_fd);
-    return NULL;
+        if (pw_fd != -1)
```

Comment:
Hello @ccoffing,

Just found this cleanup code introduced a new bug, which I've fixed in #1550. This compare should be `if (pw_fd == -1)`.

Thank you!